data: adjust extra head mesh for braid

lahm86 opened the pull request:
Resolves #1549.

#### Checklist

- [x] I have read the [coding conventions](https://github.com/LostArtefacts/TR1X/blob/master/CONTRIBUTING.md#coding-conventions)
- [x] I have added a changelog entry about what my pull request accomplishes, or it is an internal change

#### Description

This makes Lara's head in the T-rex death animation match the braid, similar to her normal head edits.

![image](https://github.com/user-attachments/assets/5ea52f94-a488-44f9-a99f-b4c80c1736a6)

lahm86 committed:
data: adjust extra head mesh for braid

This makes Lara's head in the T-rex death animation match the braid,
similar to her normal head edits.

Resolves #1549.

walkawayy commented:
Oops forgot to say do we need to mention a blurb in `gameflow.md` to guide builders?

lahm86 commented:
> Oops forgot to say do we need to mention a blurb in `gameflow.md` to guide builders?

Good point. Updated, and I've renamed the file to match the others of this type.
